cluster/gce/gci/configure-helper.sh

```diff
@@ -1453,48 +1453,76 @@ function create-master-etcd-apiserver-auth {
    fi
 }
 
+
 function docker-installed {
     if systemctl cat docker.service &> /dev/null ; then
         return 0
     else
         return 1
     fi
 }
+function addockeropt {
+	DOCKER_OPTS_FILE=/etc/docker/daemon.json
+	if [ "$#" -lt 1 ]; then
+	echo "No arguments are passed while adding docker options. Expect one argument"
+	exit 1
+	elif [ "$#" -gt 1 ]; then
+	echo "Only one argument is accepted"
+	exit 1
+	fi
+	# appends the given input to the docker opts file i.e. /etc/docker/daemon.json file
+	if [ -f "$DOCKER_OPTS_FILE" ]; then
+	cat >> "${DOCKER_OPTS_FILE}" <<EOF
```

Comment:
should we remove the option if it already present in the json? So if kubernetes attempts to add the same setting as already configured, the value will be overrriden.

Reply:
that might complicate the places where we can set multiple values like log-opt which has an internal option values.

Reply:
that might complicate for the options such as log-opt which has internal structure option value.

Reply:
I know, but this is the reason to switch to daemon.json, right? We want to make sure settings will not conlict. Will Docker be fine with multiple entries for the same setting in the daemon.json?

Reply:
Looks thing is becoming complicated using today's solution.

I would like to suggest using /etc/docker/daemon.json directly, and just 
copied it to /etc/docker when nodes boot up in the long run.

Writing script to override looks challenge, and also difficult to maintain.